src/coreclr/jit/instr.cpp

```diff
@@ -1963,6 +1963,44 @@ instruction CodeGenInterface::ins_Store(var_types dstType, bool aligned /*=false
     return ins;
 }
 
+//------------------------------------------------------------------------
+// ins_Store: Get the machine dependent instruction for performing a store to dstType on the stack from a srcReg.
+//
+// Arguments:
+//   srcReg  - the source register for the store
+//   dstType - the destination type
+//   aligned - whether the destination is properly aligned if dstType is a SIMD type
+//
+// Return Value:
+//   the instruction to use
+//
+// Notes:
+//   The fuctiuon currently does not expect float srcReg with integral dstType and will assert on such cases.
+//
+instruction CodeGenInterface::ins_Store(regNumber srcReg, var_types dstType, bool aligned /*=false*/)
```

Comment:
Would prefer not to see overloads like this with optional parameters. Can we rename one or the other?

Reply:
Also not clear to me why we don't just force all stores to use the version that does more checking...

Reply:
Makes sense, I am thinking how to rename one of them now.

> Also not clear to me why we don't just force all stores to use the version that does more checking...

It is possible but requires more refactoring changes, for example, `emitInsStoreLcl` and `emitInsStoreInd` should just receive `GenTree*` and calculate `instruction ins, emitAttr attr` inside. I will try to find time for this change later as a separate PR.